simnode/generator.py

```diff
@@ -156,44 +156,47 @@ def add_obstacles_to_scenario(self, scenario: Scenario) -> None:
                 mesh = ProceduralBump(pos, rot, obstacle.width, obstacle.length, height, obstacle.upper_length,
                                       obstacle.upper_width)
             elif obstacle_type is Stopsign:
+                rot = (0, 0, -(obstacle.z_rot - 90))
                 id_number = randrange(1000)
                 name_sign = "stopsign" + str(id_number)
-                stopsign = StaticObject(pos=(obstacle.x, obstacle.y, 0), rot=rot, name=name_sign, scale=(3, 3, 3),
+                stopsign = StaticObject(pos=(obstacle.x, obstacle.y, 0), rot=rot, name=name_sign, scale=(1.9, 1.9, 1.9),
                                         shape='/levels/drivebuild/art/objects/stopsign.dae')
                 scenario.add_object(stopsign)
             elif obstacle_type is TrafficLightSingle:
                 id_number = randrange(1000)
                 name_light = "trafficlight" + str(id_number)
                 name_pole = "pole" + str(id_number)
-                traffic_light = StaticObject(name=name_light, pos=(obstacle.x, obstacle.y, 5.32), rot=rot,
+                rot=(0, 0, -(obstacle.z_rot) - 90)
+                traffic_light = StaticObject(name=name_light, pos=(obstacle.x, obstacle.y, 4.62), rot=rot,
```

Comment:
Here you could consider hills and pits easily as well. So please consider obstacle.z as well.